### PR TITLE
Replace docker hub images with quay.io

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -280,7 +280,7 @@ spec:
       email: infinispan-dev@lists.jboss.org
   relatedImages:
     - name: infinispan-server
-      image: quay.io/infinispan/server:12.0.0.Dev06
+      image: quay.io/quay.io/infinispan/server:12.0.0.Dev06
```
Comment (pruivo) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:283]: `quay.io/quay.io` may not work :)

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -126,7 +126,7 @@ spec:
                       - name: OPERATOR_NAME
                         value: infinispan-operator
                       - name: RELATED_IMAGE_OPENJDK
-                        value: quay.io/infinispan/server:12.0.0.Dev06
+                        value: quay.io/quay.io/infinispan/server:12.0.0.Dev06
```
Comment (pruivo) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:129]: same here

File: README.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ $ oc cluster up
 $ export KUBECONFIG=/path/to/admin.kubeconfig
 ```
 3. Create the Infinispan operator on OKD from the source code.
-  * Use the public  [jboss/infinispan-operator](https://hub.docker.com/r/jboss/infinispan-operator) image:
+  * Use the public  [quay.io/infinispan/operator](https://hub.docker.com/r/quay.io/infinispan/operator) image:
```
Comment (pruivo) [inline README.md:59]: the link doesn't work

**Review by pruivo — CHANGES_REQUESTED** [review]

**Review by pruivo — APPROVED** [review]

**Comment by dmvolod** [comment]
@ryanemerson I've found some sort of regression in UpdateStatus, please wait for #673 to be tested and fixed

**Review by dmvolod — APPROVED** [review]

**Pull request merged**